yaksh/models.py

```diff
@@ -111,13 +111,17 @@ def create_trial_course(self, user):
         trial_course.enroll(False, user)
         return trial_course
 
+    def get_hidden_courses(self, code):
+        return self.filter(code=code, hidden=True)
 
 ###############################################################################
 class Course(models.Model):
     """ Course for students"""
     name = models.CharField(max_length=128)
     enrollment = models.CharField(max_length=32, choices=enrollment_methods)
     active = models.BooleanField(default=True)
+    code = models.CharField(max_length=128, null=True, blank=True)
+    hidden = models.BooleanField(default=False)
```

Comment:
Its been handled in forms where save is overridden, **hidden** is always saved as false if there is no course code, irrespective of user selection.